services/attachment/attachment.go

```diff
@@ -50,3 +50,17 @@ func UploadAttachment(ctx context.Context, file io.Reader, allowedTypes string,
 
 	return NewAttachment(ctx, attach, io.MultiReader(bytes.NewReader(buf), file), fileSize)
 }
+
+// UploadAttachmentWithCustomizableName is like UploadAttachment, but you pass both the original file name and the attachment name
+// (which could be the same).
+func UploadAttachmentWithCustomizableName(ctx context.Context, file io.Reader, allowedTypes string, fileSize int64, originalFileName string, attach *repo_model.Attachment) (*repo_model.Attachment, error) {
```

Comment:
This function has almost the exact same definition as `UploadAttachment()`. I would need to tweak a couple areas if I were to change its signature so if that's alright to do let me know!

Reply:
I think we should change `UploadAttachment` and fix all the places but not only this one.

Reply:
Changed `UploadAttachment()` and updated the affected areas 👍